src/sentry/models/useroption.py

```diff
@@ -57,21 +67,40 @@ def unset_value(self, user, project, key):
         self._option_cache[metakey].pop(key, None)
 
     def set_value(self, user, key, value, **kwargs):
+        from sentry.models.notificationsetting import NotificationSetting
+
         project = kwargs.get("project")
         organization = kwargs.get("organization")
 
         if organization and project:
             raise NotImplementedError(option_scope_error)
 
-        inst, created = self.get_or_create(
-            user=user,
-            project=project,
-            organization=organization,
-            key=key,
-            defaults={"value": value},
-        )
-        if not created and inst.value != value:
-            inst.update(value=value)
+        with transaction.atomic():
+            if key in [
+                "workflow:notifications",
+                "mail:alert",
+                "deploy-emails",
+                "subscribe_by_default",
+            ]:
+                type, option_value = get_key_value_from_legacy(key, value)
+                NotificationSetting.objects.update_settings(
+                    ExternalProviders.EMAIL,
+                    type,
+                    option_value,
+                    user=user,
+                    project=project,
+                    organization=organization,
+                )
+
+            inst, created = self.get_or_create(
```

Comment:
same here, don't you still want to update the setting regardless of which key it is? you just want the notificationsetting bit for the keys we care about

Reply:
wait I think this one is okay, hard to read indents on GH